src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -294,9 +294,16 @@ public IActionResult GetAdminToken()
         [Route("admin/warmup")]
         [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
         [RequiresRunningHost]
-        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost)
+        public async Task<IActionResult> Warmup([FromServices] WebJobsScriptHostService hostService)
```

Comment:
I don't think we should be binding to WebJobsScriptHostService and accessing its Services collection. Can we instead add an ActiveHost property to IScriptHostManager and bind to that interface?

Reply:
Also, I think we should be applying this same fix to the other 2 locations where we have this pattern (in FunctionsController)

Reply:
I'm working on the other instances. So we don't need to increase the scope of the changes here.

Reply:
In a separate PR? I'd like to see us use a consistent pattern across the board. E.g. bind to IScriptHostManager, and access an ActiveHost property (or call a TryGetActiveHost method) to get the instance.

Reply:
I like the suggestion. Exposing with the interface will help me to also be able to test this with mocks.